lisa/nic.py

```diff
@@ -151,14 +152,42 @@ def get_unpaired_devices(self) -> List[str]:
         return [x.upper for x in self.nics.values() if not x.lower]
 
     def get_upper_nics(self) -> List[str]:
-        return [x.upper for x in self.nics.values()]
+        return list(self.nics.keys())
+
+    def get_paired_devices(self) -> List[str]:
+        return [x.upper for x in self.nics.values() if x.lower]
 
     def get_lower_nics(self) -> List[str]:
         return [x.lower for x in self.nics.values() if x.lower]
 
     def get_device_slots(self) -> List[str]:
         return [x.pci_slot for x in self.nics.values() if x.pci_slot]
 
+    def get_secondary_sriov_nic(self) -> NicInfo:
+        # helper for DPDK tests where we want to pick
+        # one of the nics which isn't handling the SSH connection
+
+        default_nic_name = self.default_nic
```

Comment:
If we can sort the nics correctly, I think the two methods are not necessarily. They make the class confusing, and hard to choose.